src/buffer.jl

```diff
@@ -0,0 +1,426 @@
+# Support for python buffer protocol buffers
+
+# for versions > 3.2 (TODO: obj field not in struct for python versions <= 3.2)
+
+immutable Py_buffer
+    buf::Ptr{Void}
+    obj::PyPtr
+    len::Cssize_t
+    itemsize::Cssize_t
+    
+    readonly::Cint
+    ndim::Cint
+    format::Ptr{Cchar}
+    shape::Ptr{Cssize_t}
+    strides::Ptr{Cssize_t}
+    suboffsets::Ptr{Cssize_t}
+    internal::Ptr{Void}
+end
+ 
+type PyBuffer
+    buf::Py_buffer
+
+    PyBuffer() = begin
+    	b = new(Py_buffer(C_NULL, C_NULL, 0, 0, 
+                          0, 0, C_NULL, C_NULL,
+                          C_NULL, C_NULL, C_NULL))
+        finalizer(b, release!)
+        return b 
+    end
+end
+
+release!(b::PyBuffer) = begin
+    if b.buf.obj != C_NULL
+        ccall((@pysym :PyBuffer_Release), Void, (Ptr{PyBuffer},), &b)
+    end
+end
+
+Base.ndims(b::PyBuffer)  = int(b.buf.ndim)
+Base.length(b::PyBuffer) = b.buf.ndim >= 1 ? div(b.buf.len, b.buf.itemsize) : 0
```

Comment:
Also, is `shape` is NULL then `itemsize` should be disregarded and assumed to be `1`, according to the docs.